netbout-web/src/main/java/com/netbout/rest/Markdown.java

```diff
@@ -57,6 +58,24 @@
     private static final Tidy TIDY = Markdown.makeTidy();
 
     /**
+     * Plain link detection pattern.
+     * The regex consists of the following parts:
+     * <ul>
+     * <li>Negative lookbehind to determine we skip properly formatted
+     * links (e.g. [foo](http://bar))</li>
+     * <li>Scheme prefix followed by arbitrary number of URL-valid
+     * characters</li>
+     * <li>Last character &ndash; URL-valid characters with exclusion of common
+     * punctuation like dot, comma, question mark, exclamation mark, semicolon,
+     * colon, and parentheses</li>
+     * </ul>
+     */
+    private static final Pattern LINK = Pattern.compile(
+        // @checkstyle LineLength (1 line)
+        "(?<!(\\]\\())(https?://[a-zA-Z0-9-._~:/\\?#\\[\\]\\(\\)@!$&'*+,;=%]+[a-zA-Z0-9-_~/#\\[\\]@$&'*+=%])"
```

Comment:
@caarlos0 Simplifying it like this would cause it to grab all characters till the end of the line. So at least it will be needed to add something like positive lookahead to make sure url terminates with a space, e.g.: `(?=[ \t])`. But that would cause regex to match the text to the last space on a line, so instead of `.*` you'd need a `.*?`, to make a star non-greedy. 

As well it is probably desired to avoid url to be grabbing ending dot, semicolon, or comma (e.g. if you write link in the end of sentence, or write several links separated by comma), but adding e.g. a `.` into positive lookahead combined with non-greedy star would cause regex to terminate prematurely on a first dot inside link.